tests/generator-Tests/Unit-Tests/CodeGeneratorExpectedResults/XAJavaInterop1/WriteInterfaceFieldAsDimProperty.txt

```diff
@@ -0,0 +1,126 @@
+[Register ("com/xamarin/android/MyInterface", DoNotGenerateAcw=true)]
+public abstract class MyInterface : Java.Lang.Object {
+	internal MyInterface ()
+	{
+	}
+
+	// Metadata.xml XPath field reference: path="/api/package[@name='com.xamarin.android']/interface[@name='MyInterface']/field[@name='EGL_NATIVE_VISUAL_ID']"
+	[Register ("EGL_NATIVE_VISUAL_ID")]
+	public const int EglNativeVisualId = (int) 12334;
+
+
+	// Metadata.xml XPath field reference: path="/api/package[@name='com.xamarin.android']/interface[@name='MyInterface']/field[@name='EGL_NO_SURFACE']"
+	[Register ("EGL_NO_SURFACE")]
+	public static int EglNoSurface {
+		get {
+			const string __id = "EGL_NO_SURFACE.I";
+
+			var __v = _members.StaticFields.GetInt32Value (__id);
+			return __v;
+		}
+		set {
+			const string __id = "EGL_NO_SURFACE.I";
+
+			try {
+				_members.StaticFields.SetValue (__id, value);
+			} finally {
+			}
+		}
+	}
+
+	static readonly JniPeerMembers _members = new XAPeerMembers ("com/xamarin/android/MyInterface", typeof (MyInterface));
+
+}
+
+[Register ("com/xamarin/android/MyInterface", DoNotGenerateAcw=true)]
+[global::System.Obsolete ("Use the 'MyInterface' type. This type will be removed in a future release.", error: true)]
+public abstract class MyInterfaceConsts : MyInterface {
+	private MyInterfaceConsts ()
+	{
+	}
+
+}
+
+// Metadata.xml XPath interface reference: path="/api/package[@name='com.xamarin.android']/interface[@name='MyInterface']"
+[Register ("com/xamarin/android/MyInterface", "", "Com.Xamarin.Android.IMyInterfaceInvoker")]
+public partial interface IMyInterface : IJavaObject, IJavaPeerable {
```

Comment:
Shouldn't this interface declaration *also* contain:

```csharp
public const int EglNativeVisualId = (int) 12334;
```

I would expect the `IMyInterface` type to "mirror"/contain the same members as the `MyInterface` type, yet `IMyInterface.EglNativeVisualId` doesn't exist.

Reply:
Interface constants are gated behind `-lang-features=interface-constants`, which isn't enabled in this test suite.  

I went back and forth on this and decided that "static interface fields" falls more under `default-interface-methods` than `interface-constants`.  DIM is poorly named and really means "default AND static interface members" in this context.